lldb/tools/lldb-dap/LLDBUtils.h

```diff
@@ -33,9 +39,14 @@ namespace lldb_dap {
 /// \param[in] strm
 ///     The stream that will receive the prefix, prompt + command and
 ///     all command output.
+///
+/// \param[in] parse_command_directives
+///     If \b false, then command prefixes like \b ! or \b ? are not parsed and
+///     each command is executed verbatim.
 void RunLLDBCommands(llvm::StringRef prefix,
                      const llvm::ArrayRef<std::string> &commands,
-                     llvm::raw_ostream &strm);
+                     llvm::raw_ostream &strm,
+                     bool parse_command_directives = true);
```

Comment:
add a bool &fatal_error that is an out parameter to indicate we got an error on a ! command, clients will then use this to figure out what to do instead of aborting.